geopandas/tools/tests/test_sjoin.py

```diff
@@ -82,6 +83,41 @@ def test_sjoin_values(self):
         df = sjoin(self.polydf, self.pointdf, how='left')
         self.assertEquals(df.shape, (12,8))
 
+    def test_no_overlapping_geometry(self):
+        # Note: these tests are for correctly returning GeoDataFrame
+        # when result of the join is empty
+
+        df_inner = sjoin(self.pointdf.iloc[17:], self.polydf, how='inner')
+        df_left = sjoin(self.pointdf.iloc[17:], self.polydf, how='left')
+        df_right = sjoin(self.pointdf.iloc[17:], self.polydf, how='right')
+
+
+        empty_result_df = pd.concat([pd.Series(name='index_left',dtype='int64'),
+                                     pd.Series(name='index_right',dtype='int64')],axis=1)
+
+        expected_inner_df = pd.concat([self.pointdf.iloc[:0],
+                                       empty_result_df.index_right,
+                                       self.polydf.drop('geometry', axis = 1).iloc[:0]], axis = 1)
+
+        expected_inner = GeoDataFrame(expected_inner_df, crs = {'init': 'epsg:4326', 'no_defs': True})
+
+        expected_right_df = pd.concat([self.pointdf.iloc[17:].drop('geometry', axis = 1).iloc[:0],
```

Comment:
the `iloc[17:]` is not needed on this line I think? (since you do `iloc[:0]` at the end)